python/tvm/target.py

```diff
@@ -83,68 +77,194 @@ class Target(object):
     - :any:`tvm.target.rocm` create ROCM target
     - :any:`tvm.target.mali` create Mali target
     """
-    current = None
-
-    def __init__(self,
-                 target_name,
-                 options=None):
-        self.target_name = target_name
-        self.options = _merge_opts([], options)
-        self.device_name = ""
-        self.libs = []
-        # Parse device option
-        for item in self.options:
-            if item.startswith("-libs="):
-                libs = item.split("=")[1]
-                self.libs += libs.split(",")
-            elif item.startswith("-device="):
-                self.device_name = item.split("=")[1]
-        # Target query searches device name first
-        if self.device_name:
-            self.keys = (self.device_name,)
-        else:
-            self.keys = ()
-        # Target configuration handling
-        self.thread_warp_size = 1
-        if target_name in ("llvm", ):
-            self.keys += ("cpu",)
-        elif target_name in ("cuda", "nvptx"):
-            self.keys += ("cuda", "gpu")
-            self.max_num_threads = 512
-            self.thread_warp_size = 32
-        elif target_name in ("rocm", "opencl"):
-            # For now assume rocm schedule for opencl
-            self.keys += ("rocm", "gpu")
-            self.max_num_threads = 256
-        elif target_name in ("metal", "vulkan"):
-            self.keys += (target_name, "gpu",)
-            self.max_num_threads = 256
-        elif target_name in ("opengl",):
-            self.keys += ("opengl",)
-        elif target_name in ("stackvm", "ext_dev"):
-            # Do not now class for stackvm or ext_dev
-            pass
-        else:
-            raise ValueError("Unknown target name %s" % target_name)
-
-    def __str__(self):
-        return " ".join([self.target_name] + self.options)
-
-    def __repr__(self):
-        return self.__str__()
+
+    @property
+    def keys(self):
+        return [k.value for k in self.keys_array]
+
+    @property
```

Comment:
add unit test cases to ```tests/python/unittest/test_target.py``` to test
- parsing from string
- assert each field is correct